Support searching in events table

muddana-satish opened the pull request:
Signed-off-by: muddana-satish <satish.muddana@ericsson.com>

bhufmann reviewed (changes requested):
Overall the feature looks already good and you're on good path. I started with testing the functionality and I noticed a few things:
1- the yellow background color makes the text hard to read in dark theme (white on yellow background). The highlighting color should be different depending on the theme. For light theme the yellow is good.  In Trace Compass dark green (0x008000) works well.
2- I can't type in to search box, after clicking once into a search box. I have to click twice
![search-2-click](https://user-images.githubusercontent.com/413424/122814547-59ea6300-d2a2-11eb-8108-0715628f1339.gif)
3- When searching for a string, the highlighted text is put at the beginning of the cell text 
![wrong-replace](https://user-images.githubusercontent.com/413424/122814578-679fe880-d2a2-11eb-8943-1271a6cb6635.gif)
4-The next and close button is not visible when the column is outside the window range (not sure how to fix it). 
![image](https://user-images.githubusercontent.com/413424/122817420-ddf21a00-d2a5-11eb-88b2-468b4be926b8.png)
5-The progress wheel (when searching or reloading a new page) is shown now in all cells and not in the first column.
![table-progress-wheel](https://user-images.githubusercontent.com/413424/122817841-712b4f80-d2a6-11eb-8d75-e6ccdaf0bd08.gif)

Will continue with reviewing the code while can look at the things above.

MatthewKhouzam reviewed (commented):
In the commit message, please explain why we bump ag-grid community. Also why move fetchTableLines

muddana-satish commented:
> In the commit message, please explain why we bump ag-grid community. Also why move fetchTableLines

Updated the commit message.

muddana-satish commented:
@bhufmann. Thanks for the feedback and comments. I have updated the code that fixes few of the issues you pointed out.

> 1- the yellow background color makes the text hard to read in dark theme (white on yellow background). The highlighting color should be different depending on the theme. For light theme the yellow is good. In Trace Compass dark green (0x008000) works well.

Update the background color of the matching text in the dark theme. However for the change to reflect, the table has to be relaunched. 

> 2- I can't type in to search box, after clicking once into a search box. I have to click twice

Set the auto focus to the input field after clicking the search filter.

> 3- When searching for a string, the highlighted text is put at the beginning of the cell text

I didn't realize it's a substring match. Updated the code to highlight all the matches of the filtered text.

> 5-The progress wheel (when searching or reloading a new page) is shown now in all cells and not in the first column. Minor thing. Can be also addressed afterwards if we agree on to do so.

This change was on purpose. I felt having the spinner at all cells is a good UX.

MatthewKhouzam reviewed (approved):
Works and I like it. please address Bernd's comments. Code looks good to me.

bhufmann commented:
> @bhufmann. Thanks for the feedback and comments. I have updated the code that fixes few of the issues you pointed out.
> 
> > 1- the yellow background color makes the text hard to read in dark theme (white on yellow background). The highlighting color should be different depending on the theme. For light theme the yellow is good. In Trace Compass dark green (0x008000) works well.
> 
> Update the background color of the matching text in the dark theme. However for the change to reflect, the table has to be relaunched.
> 
That's ok for now. This is similar #385 and should be fixed together.

> > 2- I can't type in to search box, after clicking once into a search box. I have to click twice
> 
> Set the auto focus to the input field after clicking the search filter.
> 
It works now.
> > 3- When searching for a string, the highlighted text is put at the beginning of the cell text
> 
> I didn't realize it's a substring match. Updated the code to highlight all the matches of the filtered text.
> 
It works now.
> > 5-The progress wheel (when searching or reloading a new page) is shown now in all cells and not in the first column. Minor thing. Can be also addressed afterwards if we agree on to do so.
> 
> This change was on purpose. I felt having the spinner at all cells is a good UX.
I'm fine with it.

bhufmann reviewed (dismissed):
@muddana-satish Thanks for the quick updates and fixes. I noticed that the search starts always from the beginning of the trace. In the example gif below, I moved the timestamp in the table using the Resource view, then I started a search and the table gets populated from the beginning of the trace. The scrollbar of the tables indicates that. Moreover, the Resources view doesn't update it's time selection. I would expect that the search starts from the selected event (start even if a range is selected) instead of the beginning of the trace. If no event is selected, then start from the beginning of the trace.

![out-of-sync-table](https://user-images.githubusercontent.com/413424/122966099-e9e9e480-d356-11eb-8b38-6f1a92840348.gif)

PatrickTasse commented on packages/react-components/src/components/table-output-component.tsx:
```diff
@@ -331,13 +301,31 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
+                    floatingFilterComponentParams: {
+                        suppressFilterButton: true,
+                        onFilterChange: this.searchEvents,
+                        onclickNext: this.findNextMatchedEvent,
```
I expected it to find the next matched event incrementally as I type in the text box, not only after pressing Enter or clicking the next button.

It creates this strange behavior:

1. Type 'x' where x matches in that column at row 100.
The table is updated to dim the visible rows 0-10.
I expected it to go to row 100.

2. Press Enter or click Next button.
The table goes to and selects row 100.

3. Delete 'x' so that regex is empty
The table goes back to row 0 and undims all the rows.
I expected it to stay at row 100.
(note, row 100 is still selected even though it is not visible)

4. Type 'x' again
The table is updated to dim visible rows 0-10

5. Press Enter or click Next button
The table goes to and selects the next match at row 200.
This is OK since the selection was row 100, but it's confusing because the table had scrolled back to row 0.

PatrickTasse reviewed (approved):
Issues to be opened to fix later.
